app/src/main/java/com/example/topmovies/database/dao/BaseDao.kt

```diff
@@ -0,0 +1,12 @@
+package com.example.topmovies.database.dao
+
+import androidx.room.Dao
+import androidx.room.Insert
+import androidx.room.OnConflictStrategy
+
+@Dao
+interface BaseDao<T> {
+
+    @Insert(onConflict = OnConflictStrategy.IGNORE)
```

Comment:
Why have you chosen IGNORE strategy?

Reply:
I use OnConflictStrategy.IGNORE strategy because in this case, I don't need to replace all data fields. 
In this case, if I need to update part of a field, better use another query like:
 @Query("UPDATE tb_movies SET rank = :rank,rankUpDown = :rankUpDown  WHERE id = :id")
    abstract fun updateMovie(id: String, rank: String, rankUpDown: String)